src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs

```diff
@@ -104,21 +104,31 @@ public ReadOnlyBytes Slice(Range range)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public ReadOnlyBytes Slice(int index, int length)
         {
-            if (First.Length >= length + index)
+            var first = First;
+            if (first.Length >= length + index)
             {
-                return new ReadOnlyBytes(First.Slice(index, length));
+                return new ReadOnlyBytes(first.Slice(index, length));
             }
-            if (First.Length > index)
+            if (first.Length > index)
             {
-                return new ReadOnlyBytes(_first.Slice(index), _rest, length);
+                return new ReadOnlyBytes(first.Slice(index), _rest, length);
             }
```

Comment:
This used to be a member variable _first, and now it is a local variable first ( = First). Was it incorrect before?